mteb/cache.py

```diff
@@ -275,6 +279,160 @@ def download_from_remote(
 
         return results_directory
 
+    def download_cached_results_from_branch(
+        self,
+        branch: str = "cached-data",
+        filename: str = "__cached_results.json.gz",
+        output_path: Path | None = None,
+        remote: str = "https://github.com/embeddings-benchmark/results",
+        timeout: int = 60,
+        max_size_mb: int = 500,
```

Comment:
why is there a max size and timeout?

Reply:
IMO we're better off with some limits instead of none. But if you really don't like them, we can remove them.

Reply:
No - more like if the file size or the download process is out of expected nominal range. The branch can be corrupted (or hacked) to have a huge file. This way we have an added check on this side of the code as well. 
Re: timeout, here is a [relevant thread](https://github.com/psf/requests/issues/3070#issuecomment-2121302741).